File: app/src/main/java/com/altibbi/cdsSdk/MainActivity.java
package com.altibbi.cdsSdk;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    Button submitButton  ;
    EditText tokenText;
    EditText apiText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SharedPreferences sharedPreferences = getSharedPreferences("sharedPrefs",MODE_PRIVATE);
        if (!sharedPreferences.getString("TOKEN", "").isEmpty() && !sharedPreferences.getString("API_LINK", "").isEmpty()) {
            Intent i = new Intent(MainActivity.this, ApisExampleActivity.class);
            startActivity(i);
        }

        submitButton = findViewById(R.id.sdk_button);
        tokenText   = findViewById(R.id.sdk_token_text);
        apiText   = findViewById(R.id.sdk_api_text);
        submitButton.setOnClickListener(
                v -> {
                    SharedPreferences.Editor editor = sharedPreferences.edit();
                    editor.putString("TOKEN", tokenText.getText().toString());
                    editor.putString("API_LINK", apiText.getText().toString());
                    editor.apply();
                    System.out.println("ALTIBBI_SDK TOKEN ==> " + sharedPreferences.getString("TOKEN", ""));
                    System.out.println("ALTIBBI_SDK LINK ==> " + sharedPreferences.getString("API_LINK", ""));
                    Intent i = new Intent(MainActivity.this, ApisExampleActivity.class);
                    startActivity(i);
                }
        );

    }
}

File: app/src/main/java/com/altibbi/kotlinsdk/MainActivity.kt
package com.altibbi.kotlinsdk

import com.altibbi.kotlinsdk.R
import com.altibbi.kotlinsdk.consultation.ConsultationPageActivity
import com.altibbi.kotlinsdk.consultation.ConsultationListActivity
import com.altibbi.kotlinsdk.sina.AskSinaActivity
import com.altibbi.kotlinsdk.user.UserPageActivity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.button.MaterialButton

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<MaterialToolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        findViewById<MaterialButton>(R.id.button_consultation).setOnClickListener {
            startActivity(Intent(this, ConsultationPageActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.button_ask_sina).setOnClickListener {
            startActivity(Intent(this, AskSinaActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.button_consultation_list).setOnClickListener {
            startActivity(Intent(this, ConsultationListActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.button_user_page).setOnClickListener {
            startActivity(Intent(this, UserPageActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.button_sdk_setup).setOnClickListener {
            startActivity(
                Intent(this, SetupActivity::class.java)
                    .putExtra(SetupActivity.EXTRA_EDIT_CONFIG, true)
            )
            finish()
        }
    }
}
